docs/essentials/using-private-dependencies.md

```diff
@@ -78,5 +78,16 @@ That's all there is to it. You can use the approach to add more deploy
 keys to the `private-repo` secret to cover more projects and reuse the
 secret across other projects.
 
+## Using private Ruby gems
+
+If project's dependencies are hosted on Github, 
+you can enable Semaphore to fetch repositories with these steps:
+
+- Create a [machine user][machine-user] with access to repositories.
+- Create a new SSH key pair for the purpose of automation.
+- Add the private SSH key to the project as a secret.
+- Add the public SSH key to the user settings for the machine user.
```

Comment:
While reviewing this document again, I noticed that we have a dedicated paragraph for this already on line 25.

@csidyel wdyt about rewording that paragraph to be more explicit instead of adding a new section?

---

What we are missing in both the above paragraph and this new section is an answer to "why I would use a machine user?", "when is it better to use deploy keys on a single project vs machine users".

Reply:
@shiroyasha I think it would be more efficient to reword the paragraph to include this machine user guide. 

However, I lack the knowledge to answer the why and when.

Reply:
In short, you can have two scenarios:

- If you want to fetch only one gem from one repository, you can attach an SSH key to that repository.
- If you need to fetch multiple gems from multiple repositories, you either need <N> SSH keys (because Github enforces that every repo needs to have a unique keys) or to set up a machine-user with an SSH key and allow him to fetch from those N repositories.

Let me know if you need help. I'm available for chat at any point, or follow up with additional question in this thread

Reply:
Notes:

- Github has the terminology "deploy keys". These are just regular plain old SSH keys that are attached to a repository, but the terminology can be confusing.
- If you are attaching the SSH key to the repo they call it "deploy keys", if you attach them to a user they call it SSH keys.

Reply:
@shiroyasha can you check the latest commit? So I can merge

Reply:
LGTM. :shipit: